### DISCUSSION TITLE
3.0.0-beta.1

### DISCUSSION THREAD

**Discussion by mickael-menu** [readium/kotlin-toolkit#516] [Announcements]
## Resources

* [Migration guide](docs/migration-guide.md#300-beta1)
* [API Reference](https://readium.org/kotlin-toolkit/3.0.0-beta.1/)

## Changelog

:warning: Please consult [the migration guide](docs/migration-guide.md#300-beta1) to assist you in handling the breaking changes in this latest major release.

#### Navigator

* All the `completion` parameters of the `Navigator` APIs are removed.

### Changed

* All the APIs using or returning a `Date` objects are now using a custom `Instant` type. 

#### Shared

* The `Link` property key for archive-based publication assets (e.g. an EPUB/ZIP) is now `https://readium.org/webpub-manifest/properties#archive` instead of `archive`.

#### LCP

* [#493](https://github.com/readium/kotlin-toolkit/discussions/493) The LCP module does not require the Bluetooth permissions anymore to derive the device name.

#### Navigator

* [#325](https://github.com/readium/kotlin-toolkit/issues/325) Top EPUB selections no longer break when dragging the selection handles.
* Fixed applying preferences while the EPUB navigator fragment is being recreated.

## What's Changed

* Prevent Opds2Sniffer from reading the whole blob by @qnga in https://github.com/readium/kotlin-toolkit/pull/482
* Various improvements in LcpService by @qnga in https://github.com/readium/kotlin-toolkit/pull/483
* Fix various crashes by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/484
* Don't read content whose length is unknown in sniffing by @qnga in https://github.com/readium/kotlin-toolkit/pull/485
* Downgrade to jsoup 1.16.1 by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/489
* Fix opening LCP-protected publication in restricted mode by @qnga in https://github.com/readium/kotlin-toolkit/pull/494
* Fix a bunch of thread violations by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/490
* Remove `completion` from `Navigator` APIs by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/488
* Prevent reporting the current position while loading a locator by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/487
* Enable core library desugaring by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/496
* Fix various EPUB positioning issues by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/499
* Fix handling EPUB `noteref` with nested elements by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/503
* Fix broken EPUB selection in paginated mode by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/502
* Make possible crashes of PSPDFKit more explicit by @qnga in https://github.com/readium/kotlin-toolkit/pull/495
* Revamp format specifications and other changes by @qnga in https://github.com/readium/kotlin-toolkit/pull/500
* Replace `Date` with `Instant` by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/506
* Documentation by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/486
* Fix applying preferences while recreating EPUB fragments by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/507
* Change LCP device name strategy by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/508
* Support for the new 2.x LCP Profiles by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/510
* 2.4.1 by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/511
* Deprecate media1 and media2 navigators by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/512
* Rename `archive` Link property by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/514
* 3.0.0-beta.1 by @mickael-menu in https://github.com/readium/kotlin-toolkit/pull/515

**Full Changelog**: https://github.com/readium/kotlin-toolkit/compare/3.0.0-alpha.2...3.0.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/readium/kotlin-toolkit/releases/tag/3.0.0-beta.1'>3.0.0-beta.1</a>.</em>

**Comment by radusalagean** [comment]
I noticed that in beta.1, the screen flickers when attempting to jump to the previous page while selecting text.

First video is beta.1

https://github.com/readium/kotlin-toolkit/assets/11408459/1d8e75b0-03bf-47a2-9bd2-e31e8701e35a




Second video is alpha.2 (the screen doesn't flicker but the selection is broken, which is a [known issue](https://github.com/readium/kotlin-toolkit/issues/325) in WebView)


https://github.com/readium/kotlin-toolkit/assets/11408459/1ccd0361-49ec-4aad-9c08-e68d2ee9283d

**Reply by mickael-menu** [reply]
This is expected, see https://github.com/readium/kotlin-toolkit/pull/502. The WebView issue is worse than the flickering as it basically breaks the selection the user was trying to do.

Jumping to the previous/next pages while selecting text was not supported in the Kotlin toolkit.
